File: springboot-embedded-activemq/src/main/java/com/example/embeddedactivemq/Runner.java
package com.example.embeddedactivemq;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.jms.core.JmsTemplate;

@Slf4j
public class Runner implements CommandLineRunner {

  @Autowired
  JmsTemplate jmsTemplate;

  @Override
  public void run(String... args) throws Exception {
    log.info("Sending a email");
    jmsTemplate.convertAndSend("mailbox", new Email("info@example.com", "Hello"));
  }
}

File: springboot-remoting-kotlin-client/src/main/kotlin/com/example/remoting/Runner.kt
package com.example.remoting

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class Runner : CommandLineRunner {
  private val log = LoggerFactory.getLogger(Runner::class.java)

  @Autowired
  private var jmsInvoker: DnsService? = null

  @Scheduled(fixedRate = 1000L)
  fun query() {
    log.info("execute query...")
    log.info(">>>>>>>>>>>>>>>. {}", jmsInvoker.toString())
    log.info("[JMS         ] DNS Result: {}", jmsInvoker?.getName(""))
  }

  override fun run(vararg args: String?) {
    query()
//    log.info("[JMS         ] DNS Result: {}", jmsInvoker?.getName(""))
  }
}
